dynamic_clamp/dynamic_clamp.ino

```diff
@@ -31,11 +31,49 @@
 //            and inactivation numbers for a range of membrane potentials and stores them in the lookup table variables,
 //            and the function called at every time step to calculate the sodium current to be injected. 
 //
-// Last revised 05/20/2018, 7:00 pm - ND, CR
 
 
 #include <math.h>
 
+#if defined(TEENSYDUINO)
+    // => running on teensy 3.6
+    #define DIGITAL_READ(pin) digitalReadFast(pin)
+    #define SEND_NOW() Serial.send_now()
+    // ADC pin used to read membrane potential
+    #define PIN_ANALOG_IN 0
+    // DAC pin used to output current
+    #define PIN_ANALOG_OUT A21
+    // pin used to trigger EPSC train
+    #define PIN_EPSC_TRIGGER 2
+
+#elif defined(ADAFRUIT_ITSYBITSY_M4_EXPRESS)
+    // => running on adafruit itsybitsy m4
+    // `elapsedMillis` library is a required dependency
+    #include <elapsedMillis.h>
+    #define DIGITAL_READ(pin) digitalRead(pin)
+    #define SEND_NOW() Serial.flush()
+    // ADC pin used to read membrane potential
+    #define PIN_ANALOG_IN A3
+    // DAC pin used to output current
+    #define PIN_ANALOG_OUT DAC0
+    // pin used to trigger EPSC train
+    #define PIN_EPSC_TRIGGER 7
+
+#else
+    // => running on some other board

```

Comment:
using defines to automatically figure out what kind of board is being targeted by the build and adjusting configuration and functions as necessary.  Teensy modifications to Arduino provide some features that are not available on other boards - this keeps the main code the same while providing easy configuration.